[2023-02-03T12:43:48.856+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: spark-test.spark_job manual__2023-02-03T12:43:46.249994+00:00 [queued]>
[2023-02-03T12:43:48.880+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: spark-test.spark_job manual__2023-02-03T12:43:46.249994+00:00 [queued]>
[2023-02-03T12:43:48.881+0000] {taskinstance.py:1283} INFO - 
--------------------------------------------------------------------------------
[2023-02-03T12:43:48.883+0000] {taskinstance.py:1284} INFO - Starting attempt 1 of 2
[2023-02-03T12:43:48.884+0000] {taskinstance.py:1285} INFO - 
--------------------------------------------------------------------------------
[2023-02-03T12:43:48.917+0000] {taskinstance.py:1304} INFO - Executing <Task(SparkSubmitOperator): spark_job> on 2023-02-03 12:43:46.249994+00:00
[2023-02-03T12:43:48.931+0000] {standard_task_runner.py:55} INFO - Started process 2108 to run task
[2023-02-03T12:43:48.939+0000] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'spark-test', 'spark_job', 'manual__2023-02-03T12:43:46.249994+00:00', '--job-id', '16', '--raw', '--subdir', 'DAGS_FOLDER/spark.py', '--cfg-path', '/tmp/tmpzrfyoxgo']
[2023-02-03T12:43:48.941+0000] {standard_task_runner.py:83} INFO - Job 16: Subtask spark_job
[2023-02-03T12:43:49.139+0000] {task_command.py:389} INFO - Running <TaskInstance: spark-test.spark_job manual__2023-02-03T12:43:46.249994+00:00 [running]> on host airflow-worker
[2023-02-03T12:43:49.426+0000] {taskinstance.py:1513} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_EMAIL=airflow@airflow.com
AIRFLOW_CTX_DAG_OWNER=airflow
AIRFLOW_CTX_DAG_ID=spark-test
AIRFLOW_CTX_TASK_ID=spark_job
AIRFLOW_CTX_EXECUTION_DATE=2023-02-03T12:43:46.249994+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=manual__2023-02-03T12:43:46.249994+00:00
[2023-02-03T12:43:49.462+0000] {spark_submit.py:223} INFO - Could not load connection string spark_default, defaulting to yarn
[2023-02-03T12:43:49.467+0000] {spark_submit.py:344} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name Spark Hello World --verbose /opt/airflow/plugins/app.py
[2023-02-03T12:43:49.710+0000] {spark_submit.py:495} INFO - /home/airflow/.local/lib/python3.7/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2023-02-03T12:43:49.714+0000] {spark_submit.py:495} INFO - JAVA_HOME is not set
[2023-02-03T12:43:49.757+0000] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 427, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name Spark Hello World --verbose /opt/airflow/plugins/app.py. Error code is: 1.
[2023-02-03T12:43:49.766+0000] {taskinstance.py:1327} INFO - Marking task as UP_FOR_RETRY. dag_id=spark-test, task_id=spark_job, execution_date=20230203T124346, start_date=20230203T124348, end_date=20230203T124349
[2023-02-03T12:43:49.819+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 16 for task spark_job (Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --name Spark Hello World --verbose /opt/airflow/plugins/app.py. Error code is: 1.; 2108)
[2023-02-03T12:43:49.877+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2023-02-03T12:43:49.952+0000] {taskinstance.py:2582} INFO - 0 downstream tasks scheduled from follow-on schedule check
